Hi Sadie - nearly there :) `test_regrid.py` still has  a bug:
```python
test_Field_regrid_2d_field (__main__.RegridTest.test_Field_regrid_2d_field)
2-d regridding with Field destination grid. ... ERROR
test_Field_regrid_chunks (__main__.RegridTest.test_Field_regrid_chunks)
Regridding of chunked axes ... ok
test_Field_regrid_weights_file (__main__.RegridTest.test_Field_regrid_weights_file)
Regridding creation/use of weights file ... ERROR
test_Field_regridc_1d_coordinates_z (__main__.RegridTest.test_Field_regridc_1d_coordinates_z)
1-d Z Cartesian regridding with coordinates destination grid. ... ERROR
test_Field_regridc_1d_field (__main__.RegridTest.test_Field_regridc_1d_field)
1-d Cartesian regridding with Field destination grid. ... ERROR
test_Field_regridc_2d_coords (__main__.RegridTest.test_Field_regridc_2d_coords)
2-d Cartesian regridding with coords destination grid. ... ERROR
test_Field_regridc_3d_field (__main__.RegridTest.test_Field_regridc_3d_field)
3-d Cartesian regridding with Field destination grid. ... ERROR
test_Field_regridc_domain (__main__.RegridTest.test_Field_regridc_domain)
Spherical regridding with Domain destination grid. ... ERROR
test_Field_regrids_bad_dst (__main__.RegridTest.test_Field_regrids_bad_dst)
Disallowed destination grid types raise an exception. ... ERROR
test_Field_regrids_coords (__main__.RegridTest.test_Field_regrids_coords)
Spherical regridding with coords destination grid. ... ERROR
test_Field_regrids_domain (__main__.RegridTest.test_Field_regrids_domain)
Spherical regridding with Domain destination grid. ... ERROR
test_Field_regrids_field_operator (__main__.RegridTest.test_Field_regrids_field_operator)
Spherical regridding with operator destination grid. ... ERROR
test_Field_regrids_non_coordinates (__main__.RegridTest.test_Field_regrids_non_coordinates)
Check setting of non-coordinate metadata. ... ERROR
test_return_esmpy_regrid_operator (__main__.RegridTest.test_return_esmpy_regrid_operator)
esmpy regrid operator returns esmpy.Regrid in regrids and regridc ... ERROR

======================================================================
ERROR: test_Field_regrid_2d_field (__main__.RegridTest.test_Field_regrid_2d_field)
2-d regridding with Field destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 166, in test_Field_regrid_2d_field
    dst = self.dst.copy()
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regrid_weights_file (__main__.RegridTest.test_Field_regrid_weights_file)
Regridding creation/use of weights file
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 783, in test_Field_regrid_weights_file
    dst = self.dst
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regridc_1d_coordinates_z (__main__.RegridTest.test_Field_regridc_1d_coordinates_z)
1-d Z Cartesian regridding with coordinates destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 755, in test_Field_regridc_1d_coordinates_z
    src = cf.read(self.filename_xyz)[0]
                  ^^^^^^^^^^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'filename_xyz'

======================================================================
ERROR: test_Field_regridc_1d_field (__main__.RegridTest.test_Field_regridc_1d_field)
1-d Cartesian regridding with Field destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 669, in test_Field_regridc_1d_field
    dst = self.dst.copy()
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regridc_2d_coords (__main__.RegridTest.test_Field_regridc_2d_coords)
2-d Cartesian regridding with coords destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 386, in test_Field_regridc_2d_coords
    dst = self.dst.copy()
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regridc_3d_field (__main__.RegridTest.test_Field_regridc_3d_field)
3-d Cartesian regridding with Field destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 565, in test_Field_regridc_3d_field
    dst = self.dst.copy()
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regridc_domain (__main__.RegridTest.test_Field_regridc_domain)
Spherical regridding with Domain destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 452, in test_Field_regridc_domain
    dst = self.dst
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regrids_bad_dst (__main__.RegridTest.test_Field_regrids_bad_dst)
Disallowed destination grid types raise an exception.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 417, in test_Field_regrids_bad_dst
    self.src.regrids(999, method="conservative")
    ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'src'

======================================================================
ERROR: test_Field_regrids_coords (__main__.RegridTest.test_Field_regrids_coords)
Spherical regridding with coords destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 317, in test_Field_regrids_coords
    dst = self.dst.copy()
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regrids_domain (__main__.RegridTest.test_Field_regrids_domain)
Spherical regridding with Domain destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 429, in test_Field_regrids_domain
    dst = self.dst
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regrids_field_operator (__main__.RegridTest.test_Field_regrids_field_operator)
Spherical regridding with operator destination grid.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 477, in test_Field_regrids_field_operator
    dst = self.dst
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

======================================================================
ERROR: test_Field_regrids_non_coordinates (__main__.RegridTest.test_Field_regrids_non_coordinates)
Check setting of non-coordinate metadata.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 515, in test_Field_regrids_non_coordinates
    src = self.src
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'src'

======================================================================
ERROR: test_return_esmpy_regrid_operator (__main__.RegridTest.test_return_esmpy_regrid_operator)
esmpy regrid operator returns esmpy.Regrid in regrids and regridc
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/david/cf-python/cf/test/test_regrid.py", line 818, in test_return_esmpy_regrid_operator
    dst = self.dst
          ^^^^^^^^
AttributeError: 'RegridTest' object has no attribute 'dst'

----------------------------------------------------------------------
Ran 14 tests in 3.044s

FAILED (errors=13)
```

--- a/cf/test/test_regrid.py
+++ b/cf/test/test_regrid.py
@@ -146,18 +146,18 @@ class RegridTest(unittest.TestCase):
             self.skipTest(
                 "Test module requires 'esmpy' package. Install it to run all."
             )
-        else:
-            # Get the test source and destination fields
-            filename = os.path.join(
-                os.path.dirname(os.path.abspath(__file__)), "regrid.nc"
-            )
-            dst_src = cf.read(filename)
-            dst = dst_src[0]  # noqa: F841
-            src = dst_src[1]  # noqa: F841
 
-            filename_xyz = os.path.join(  # noqa: F841
-                os.path.dirname(os.path.abspath(__file__)), "regrid_xyz.nc"
-            )
+        # Get the test source and destination fields
+        filename = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)), "regrid.nc"
+        )
+        dst_src = cf.read(filename)
+        self.dst = dst_src[0]  # noqa: F841
+        self.src = dst_src[1]  # noqa: F841
+
+        self.filename_xyz = os.path.join(  # noqa: F841
+            os.path.dirname(os.path.abspath(__file__)), "regrid_xyz.nc"
+        )
 
     def test_Field_regrid_2d_field(self):
         """2-d regridding with Field destination grid."""

--- a/cf/test/test_regrid.py
+++ b/cf/test/test_regrid.py
@@ -152,8 +152,8 @@ class RegridTest(unittest.TestCase):
             os.path.dirname(os.path.abspath(__file__)), "regrid.nc"
         )
         dst_src = cf.read(filename)
-        self.dst = dst_src[0]  # noqa: F841
-        self.src = dst_src[1]  # noqa: F841
+        self.dst = dst_src[0]
+        self.src = dst_src[1]
 
         self.filename_xyz = os.path.join(  # noqa: F841
             os.path.dirname(os.path.abspath(__file__)), "regrid_xyz.nc"
@@ -626,7 +626,7 @@ class RegridTest(unittest.TestCase):
                 #    # effectively) 1-d regridding
                 #    continue
 
-                if method in ("nearest_dtos"):
+                if method in ("nearest_dtos",):
                     continue
 
                 x = src.regridc(

